src/Model/Order/Order.php

```diff
@@ -64,16 +83,46 @@ public function getContactPerson(): NonNullContactPerson
         return $this->contactPerson;
     }
 
-    public function getHasPaid(): NonNullOrderIsPaid
+    public function getPaymentStatus(): NonNullPaymentStatus
     {
-        return $this->hasPaid;
+        return $this->paymentStatus;
     }
 
     public function getAmount(): NonNullOrderAmount
     {
         return $this->amount;
     }
 
+    public function getDatePaid(): DatePaid
+    {
+        return $this->datePaid;
+    }
+
+    public function getPaymentId(): NonNullEntityId
```

Comment:
I would say some of these payment properties should be nullable e.g. just `EntityId`, `PaymentDate` etc. 

I don't fully know the domain so I might be wrong. But if an order can exist without being paid then these properties would have a null value object.

Reply:
Again, not 100% sure on domain but it seems that it would make more sense to have a `payments` property which is an `EntitySet` containing each of the payments. In this case my original comment is redundant and they make more sense as being `NonNull...`